query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/QueryResource.scala

```diff
@@ -249,36 +249,32 @@ class QueryResource(secondary: Secondary,
          */
         def possiblyRewriteOneAnalysisInQuery(schema: Schema, analyzedQuery: Seq[SoQLAnalysis[String, SoQLType]])
           : (Seq[SoQLAnalysis[String, SoQLType]], Option[String]) = {
-          analyzedQuery.foldLeft((Seq.empty[SoQLAnalysis[String, SoQLType]], None: Option[String])) {
-            (acc, anal) =>
-              val existingRollupName = acc._2
-              existingRollupName match {
-                case Some(_) => (acc._1 :+ anal, existingRollupName)
-                case None =>
-                  val (rewrittenAnal, rollupName) = possiblyRewriteQuery(schema, anal)
-                  rollupName match {
-                    case Some(_) => (acc._1 :+ rewrittenAnal, rollupName)
-                    case None => (acc._1 :+ anal, None)
-                  }
-              }
-          }
-        }
-
-        def possiblyRewriteQuery(schema: Schema, analyzedQuery: SoQLAnalysis[String, SoQLType])
-          : (SoQLAnalysis[String, SoQLType], Option[String]) = {
           if (noRollup) {
             (analyzedQuery, None)
           } else {
             rollupInfoFetcher(base.receiveTimeoutMS(schemaTimeout.toMillis.toInt), dataset, copy) match {
               case RollupInfoFetcher.Successful(rollups) =>
-                val rewritten = RollupScorer.bestRollup(
-                  queryRewriter.possibleRewrites(schema, analyzedQuery, rollups).toSeq)
-                val (rollupName, analysis) = rewritten map { x => (Some(x._1), x._2) } getOrElse ((None, analyzedQuery))
-                if (rollupName.isDefined) {
-                  log.info(s"Rewrote query on dataset $dataset to rollup $rollupName")
+                val acc = analyzedQuery.foldLeft((Seq.empty[SoQLAnalysis[String, SoQLType]], None: Option[String], Map.empty[String, String])) {
+                  (acc, anal) =>
+                    val existingRollupName = acc._2
```

Comment:
I'd generally prefer to completely destructure `acc` here, something like `val (aThing, existingRollupName, anotherThing) = acc`